src/components/ReceiptModal/ReceiptModal.jsx

```diff
@@ -0,0 +1,223 @@
+    const [loadingPercent, setLoadingPercent] = useState(0)
+    const [receiptItems, setReceiptItems] = useState([])
+    const [uploadingImage, setUploadingImage] = useState(false)
+    const [scanningImage, setScanningImage] = useState(false)
+    const { userProducts, setUserProducts, ...modalProps } = props
+
+    useEffect(() => {
+        if (curImageFile != null) {
+            setUploadingImage(true)
+            const imageRef = ref(storage, `images/${curImageFile.name}`)
+            const uploadTask = uploadBytesResumable(imageRef, curImageFile);
+            uploadTask.on(
+                "state_changed",
+                (snapshot) => {
+                    const percent = Math.round(
+                        (snapshot.bytesTransferred / snapshot.totalBytes) * 100
+                    );
+
+                    // update progress
+                    setLoadingPercent(percent);
+                },
+                (err) => console.log(err),
+                () => {
+                    // download url
+                    getDownloadURL(uploadTask.snapshot.ref).then((url) => {
+                        console.log(url);
+                        setCurImageUrl(url)
+                    });
+                    setUploadingImage(false)
+                }
+            );
+        }
+    }, [curImageFile])
+
+    const onImageChange = (event) => {
+        let img = event.target.files[0];
+        setCurImageFile(img)
+    }
+
+    const handleReceiptRemove = () => {
+        setCurImageUrl("")
+        setCurImageFile(null)
+        setLoadingPercent(0)
+    }
+
+    const handleScanReceipt = async () => {
+        try {
+            setScanningImage(true)
+            let imageObject = {
+                url: curImageUrl
+            }
+            await axios.post(receiptScannerUrl, imageObject).then((res) => {
+                console.log(res.data)
+                const curItems = res.data.Items.values.map((val) => {
+                    let item = {
+                        name: val?.properties?.Description?.value || "",
+                        quantity: val?.properties?.Quantity?.value || "",
+                        unit: basicUnits[0],
+                        category: basicCategories[0]
+                    }
+                    return item
+                })
+                setReceiptItems(curItems)
+            }).catch((error) => {
+                console.log(error.message)
+            })
+        } catch (error) {
+            console.log(error.message)
+        }
+        setScanningImage(false)
+    }
+
+    const handleReceiptOutputFormChange = (event, indx) => {
+        const curKey = event.target.name
+        const newItems = receiptItems.map(i => ({ ...i }))
+        newItems[indx][curKey] = event.target.value
+        setReceiptItems(newItems)
+    }
+
+    const handleEachItemUpdate = (item, products) => {
+        let itemIndex = products.findIndex(val => val.name === item.name)
```

Comment:
Since you're not changing itemIndex it can be const itemIndex instead of let itemIndex?